### PR TITLE
Introduce the keep verb

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

Replace the non documented `--clean -binaries`

### Motivation

Add a quicker way to plays with the options.

### CODE REVIEW

File: cmd/cli/cmd.go [hunk 1]
```diff
@@ -316,6 +316,9 @@ func NewCommand() (*cobra.Command, *int) {
 	daemonCommand.PersistentFlags().StringP("clean", "c", config.ViperConfig.GetString("clean"), fmt.Sprintf("clean options before %s: %s", setupCommand.Name(), options.GetOptionNames(options.Clean{})))
 	config.ViperConfig.BindPFlag("clean", daemonCommand.PersistentFlags().Lookup("clean"))
 
+	daemonCommand.PersistentFlags().StringP("keep", "k", config.ViperConfig.GetString("keep"), fmt.Sprintf("clean everything but the given options before %s: %s", setupCommand.Name(), options.GetOptionNames(options.Clean{})))
```
Comment (devonboyer) [inline cmd/cli/cmd.go]: What is the behaviour of `keep` if the setup command has never been called? Just a no-op?

File: pkg/options/options.go [hunk 2]
```diff
@@ -89,36 +66,36 @@ func newOptions(stringOptions string, opt interface{}) interface{} {
...
 	for i := range availableOptions {
 		availableOptions[i] = strings.ToLower(availableOptions[i])
 	}
 	for _, elt := range strings.Split(stringOptions, ",") {
 		switch elt {
 		case "all":
-			setAll(opt)
+			setAllOptionsTo(opt, value)
+			reflect.ValueOf(opt).Elem().FieldByName("All").SetBool(value)
+			reflect.ValueOf(opt).Elem().FieldByName("None").SetBool(!value)
```
Comment (devonboyer) [inline pkg/options/options.go]: I'm not a huge fan of inspecting structs in this way but I do understand why you needed to do it in this case. Even if there was a way to refactor this it should probably be in a different PR, so I'll let it go for now.

File: pkg/options/clean.go [hunk 3]
```diff
@@ -31,10 +31,14 @@ type Clean struct {
 	Logs      bool `json:"logs,omitempty"`
 }
 
-// NewCleanOptions instantiate a new Clean from the cleanString
+// NewCleanOptions instantiate a new Clean from the cleanString and keepString
 // The clean string is lowercase clean options comma separated like: etcd,binaries ...
-func NewCleanOptions(cleanString string) *Clean {
-	return newOptions(cleanString, &Clean{}).(*Clean)
+// keepString takes precedence over the clean one
```
Comment (devonboyer) [inline pkg/options/clean.go]: Can we add something like `keepString takes precedence over the clean one` in the docs for the `keep` command?

File: cmd/cli/cmd.go (hunk 1 above)
Comment (JulienBalestra) [reply cmd/cli/cmd.go]: It's always called by `clean`.

As the workflow is `clean` -> `setup` -> `run`.

File: pkg/options/options.go (hunk 2 above)
Comment (JulienBalestra) [reply pkg/options/options.go]: I know it's awful. I'm up for any improvement.

File: pkg/options/options.go [hunk 4]
```diff
@@ -77,7 +54,7 @@ func GetOptionNames(opt interface{}) string {
 	return strings.Join(names, ",")
 }
 
-func newOptions(stringOptions string, opt interface{}) interface{} {
+func newOptions(stringOptions string, value bool, opt interface{}) interface{} {
```
Comment (devonboyer) [inline pkg/options/options.go]: Can you rename the local variable `value` to `drain/clean` maybe (since that is what `true` indicates)?

Comment (JulienBalestra) [reply pkg/options/options.go]: What do you think of something like `enabled`?

File: pkg/options/clean.go (hunk 3 above)
Comment (JulienBalestra) [reply pkg/options/clean.go]: addressed

File: pkg/options/options.go (hunk 4 above)
Comment (devonboyer) [reply pkg/options/options.go]: I like that

Comment (JulienBalestra) [reply pkg/options/options.go]: @devonboyer I just pushed the changes

**Pull request merged**